app/script/components/ephemeralTimer.js

```diff
@@ -29,16 +29,21 @@ z.components.EphemeralTimer = class EphemeralTimer {
     const dashLength = 12.6;
     const dial = componentInfo.element.querySelector('.ephemeral-timer__dial');
 
+    const numberOfAnimationSteps = 40;
+    const animationRate = duration / numberOfAnimationSteps;
     const animatePie = () => {
       const remainingTime = messageEntity.ephemeral_expires() - Date.now();
-      dial.style.strokeDashoffset = dashLength - (remainingTime / duration) * -dashLength;
-      this.animationFrameCallback = window.requestAnimationFrame(animatePie);
+      const newDashoffset = dashLength - (remainingTime / duration) * -dashLength;
+      dial.style.strokeDashoffset = Math.max(newDashoffset, dashLength);
+      if (newDashoffset >= dashLength) {
+        this.animationTimeout = window.setTimeout(animatePie, animationRate);
+      }
     };
-    this.animationFrameCallback = window.requestAnimationFrame(animatePie);
+    this.animationTimeout = window.setTimeout(animatePie, animationRate);
```

Comment:
Do we really want to set timeouts for this? I suppose this could break for long durations. Every timeout greater then 24.5 days will execute immediately. So it should event fit with the max of 1 year. But in the long run? Definitely a potential place of failure that we tried to get rid of during this sprint.

Reply:
Let's add a upper threshold to prevent this and we are good.